edivorce/apps/core/middleware/basicauth_middleware.py

```diff
@@ -6,14 +6,26 @@
 from django.template.loader import render_to_string
 
 
-class BasicAuthMiddleware(object):
+class BasicAuthMiddleware:
     """
     Simple Basic Authentication module to password protect test environments
     based on : https://djangosnippets.org/snippets/2468/
     This could have also been implemented via the NGINX-PROXY, but a Django
     implementation allows environment variables to be used to store
     username + password
     """
+    def __init__(self, get_response=None):
+        self.get_response = get_response
+        super().__init__()
+
+    def __call__(self, request):
+        response = None
+        if hasattr(self, 'process_request'):
+            response = self.process_request(request)
+        if not response:
+            response = self.get_response(request)
+        return response
+
```

Comment:
Any objections to using django.utils.deprecation.MiddlewareMixin which encapsulates the middleware style change?

Reply:
I was torn about it. 
Given the package location, that it was meant as a stop gap solution to handle transitioning to a new way of writing middlewares. But I also don't think that the way I've written it adheres to the new-style middlewares anyways. 

[Current middleware documentation](https://docs.djangoproject.com/en/3.0/topics/http/middleware/#writing-your-own-middleware)
If you have ideas about how to make it more current, I am happy to change it. Otherwise, I can make it use the mixin instead.

Reply:
Ya its strange that they have it labelled in deprecation considering that class based middlewares are still valid. Personally I think the mixin approach is fine.